core/objects.go

```diff
@@ -254,6 +274,10 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord *ValidationRecord `json:"validationRecord,omitempty"`
```

Comment:
For *most* validation types, an array doesn't make sense. For instance, what would an array mean for a DNS challenge? What would it mean for an email challenge, if that ever became part of the spec? That's why it's nice to be able to define different validation types for different challenge types: You can define the structure that makes sense for the specific challenge.